src/services/review/RepoCheckerService.ts

```diff
@@ -0,0 +1,235 @@
+                // Amplify supports adding of trailing slash
+                !setOfAllDocumentsAndPagesPath.has(`${encodedHref}/`)
+              ) {
+                console.log(setOfAllDocumentsAndPagesPath)
+                this.persistLogs(repo, [
+                  new SiteCheckerError(
+                    `Broken link ${href} that links ${text} found in ${fileName} which has permalink ${filePermalink}`
+                  ),
+                ])
+              }
+            }
+          })
+          const imgTags = dom.window.document.querySelectorAll("img")
+          forEach(imgTags, (tag) => {
+            const src = tag.getAttribute("src")
+            if (src) {
+              const encodedSrc = this.documentPathDecoder(src)
+              if (
+                !this.isExternalLinkOrPageRef(encodedSrc) &&
+                !setOfAllDocumentsAndPagesPath.has(encodedSrc)
+              ) {
+                this.persistLogs(repo, [
+                  new SiteCheckerError(
+                    `Broken image ${src} found in ${fileName} which has permalink ${filePermalink}`
+                  ),
+                ])
+              }
+            }
+          })
+        })
+        console.log("Site checker done")
+        return ok(undefined)
+      })
+    try {
+      console.log(await checker)
+    } catch (error) {
+      console.log(error)
+    }
+  }
+
+  isExternalLinkOrPageRef = (link: string) =>
+    link.startsWith("http") ||
+    link.startsWith("www") ||
+    link.startsWith("mailto:") ||
+    link.startsWith("#")
+
+  documentPathDecoder = (documentPath: string) => {
+    for (let i = 0; i < ALLOWED_FILE_EXTENSIONS.length; i += 1) {
+      if (documentPath.endsWith(ALLOWED_FILE_EXTENSIONS[i])) {
+        return decodeURIComponent(documentPath)
+      }
+    }
+    return documentPath
+  }
+
+  getListOfMarkdownFiles(
+    repoPath: string
+  ): ResultAsync<Set<string>, SiteCheckerError> {
+    const setOfAllMarkdownFiles = new Set<string>()
+    return fromPromise(
+      fs.promises.readdir(repoPath, { recursive: true }),
+      (error) => new SiteCheckerError(`${error}`)
+    ).andThen((files) => {
+      files
+        .filter((file) => !file.startsWith(".") && file.endsWith(".md"))
+        .map((file) => {
+          setOfAllMarkdownFiles.add(file)
+          return ok(undefined)
+        })
+      return ok(setOfAllMarkdownFiles)
+    })
+  }
+
+  persistLogs(repo: string, errors: SiteCheckerError[]) {
+    // create a log file in the root dir of all repos, in case of multiple errors
+    const logFilePath = path.join(process.cwd(), `log.txt`)
+
+    const logFile = fs.createWriteStream(logFilePath, { flags: "a" })
+    forEach(errors, (error) => {
+      logFile.write(`${error}\n`)
+    })
+    logFile.end()
+  }
+
+  emailErrors(errors: RepoError[]): ResultAsync<undefined, Error> {
+    // todo: send email to admin@isomer.gov.sg
+    return errAsync(new Error("Not implemented"))
+  }
+
+  getAllDocumentsPath(
```

Comment:
nit: I think we generally use "media" instead of "documents" to represent files and images, maybe we should use that?

Reply:
fair